lib/cylc/scheduler.py

```diff
@@ -1123,9 +1123,8 @@ def configure_pyro(self):
     def load_suiterc(self, reconfigure):
         """Load and log the suite definition."""
 
-        old_config = False
         if reconfigure:
-            old_config = self.config
+            old_tasks = set(self.config.get_task_name_list())
```

Comment:
I requested the removal of this. Primarily it's just better to only grab the bit needed rather than request additional memory allocation for stuff that isn't actually required. In theory you'd get an increased memory footprint here of N + N if you make a copy of the original vs. N + M where N > M for using only the subset of information needed (a list of names rather than their full configs).

I'd also be concerned about further memory spikes during the reload due to keeping the full extra config around. I think mostly there wouldn't be a problem but I pondered whether some of the larger, nastier suites that are starting to appear might be awkward.

TLDR version - probably won't make a noticeable difference but is just a better way to go about it.